processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -284,10 +284,16 @@ else if ( !method.isUpdateMethod() ) {
 
             initializeMappingReferencesIfNeeded( resultTypeToMap );
 
-            // map properties with mapping
-            boolean mappingErrorOccurred = handleDefinedMappings( resultTypeToMap );
-            if ( mappingErrorOccurred ) {
-                return null;
+            boolean hasSubclassMappingsWithAllSupportedClasses = hasSubclassMappingsWithAllSupportedClasses(
+                resultTypeToMap );
+
+
+            if ( hasSubclassMappingsWithAllSupportedClasses ) {
```

Comment:
Maybe there is another way to check if the `SubclassExhaustiveStrategy.RUNTIME_EXCEPTION` option is set? Maybe via `isAbstractReturnTypeAllowed()`?

I mean, perhaps this method should only be called, when the `SubclassExhaustiveStrategy.RUNTIME_EXCEPTION` is set, and not in every bean mapping method? This could have side effects in future code changes.

e.g.:
```java
if ( isRuntimeExceptionSubclassExhaustiveStrategy
     && hasSubclassMappingsWithAllSupportedClasses(...) ) {
  // ...
}
```

Reply:
You are right. It didn't quite feel right to use this naming. I'll think of a better solution for this

Reply:
I've renamed this. Is it clearer now?

Reply:
Going back a bit through some of the issues and my commits, I realized why I named it like I did. I am using that in https://github.com/filiphr/mapstruct/commit/d5cc43a19552967c7c682b9a1daec994a771445d which is a fix for #3360. @thunderhook do you think that the new local variable name is clear for that other commit as well?